channels/http.py

```diff
@@ -120,8 +120,11 @@ def __init__(self, scope, body):
         # TODO: chunked bodies
 
         # Limit the maximum request data size that will be handled in-memory.
+        # but only for non-multipart requests, because files are handled
+        # differently by django, see #1240
         if (
-            settings.DATA_UPLOAD_MAX_MEMORY_SIZE is not None
+            self.content_type != "multipart/form-data"
+            and settings.DATA_UPLOAD_MAX_MEMORY_SIZE is not None
```

Comment:
I'm wondering about the security implications that could relate to this change. Does this mean that multipart/form-data requests will never have any control on their size at all, and if so could that become a vector for DoS attacks ? What is DATA_UPLOAD_MAX_MEMORY_SIZE really for if we can bypass it by changing the content type ?

Reply:
@jpic, based on [Django's implementation](https://github.com/django/django/blob/77d25dbd0f20d6a907c805ffae8aaadd87edbacf/django/http/multipartparser.py#L179), which is called when accessing `request.POST`, there is a check. The second test check I've added in my PR should also ensure we never have a regression at that level.

Reply:
Thanks for baring with me, your explanation helps but I still struggle to understand how this implementation maintains the DATA_UPLOAD_MAX_MEMORY_SIZE check against non-boundary request data at all in the case of a multipart/form-data request.

Are you trying to:

A. completely disable DATA_UPLOAD_MAX_MEMORY_SIZE check in the case of multipart form data, or 
B. ensure boundary parts of the requests are excluded from the DATA_UPLOAD_MAX_MEMORY_SIZE check against the rest of the request ?

Perhaps you could add a test where there is a multipart request and a user-agent string that would be larger than DATA_UPLOAD_MAX_MEMORY_SIZE ?

Hope that makes sense otherwise please let me know i will try to reword